extension/test/test-logger.js

```diff
@@ -10,7 +10,13 @@ var verifyAsyncOutput = function (test, logger, testFunc) {
   var message = '';
 
   return function (status) {
-    var line = {}, hasMore;
+    var line = {}, hasMore, isSuccessStatus;
+
+    isSuccessStatus = typeof status == "number";
+    test.assert(isSuccessStatus, status["message"] ? status.message : status);
+    if (!isSuccessStatus) {
+      return;
+    }
```

Comment:
> Isn't `test.assert` an instance of the assertion class? Shouldn't you use a specific test method here?

Are you asking that the `test` object in these tests are equal to the SDK [`assert`][1] ?
No, they aren't. These `test` objects are [`TestRunner`][2] instances.

Or I don't understand your question. :-/

> It would be also good to add at least a one-liner for documentation what those lines are doing.

Agree.

I'd like to test here the write status. If it isn't a number (according to it's [specification][3]) then the write was unsuccessful.
The `status.message` part is just a hack to provide some details. As I wrote previously in my other comments, in that case `status` is mostly an Error instance, so it would have a message attribute.

Of course it would be nice to not to abuse test frameworks as error reporters. :)
So keeping simple the test message as "The type of the write status should be number!" would be nice. :)

Instead, if `status` is really an Error, then we could log it to console just before the tests returns early, in `if (!isSuccessStatus) { ...` case.


[1]: https://developer.mozilla.org/en-US/Add-ons/SDK/Low-Level_APIs/test_assert
[2]: https://github.com/mozilla/addon-sdk/blob/1.17/lib/sdk/deprecated/unit-test.js#L43-L489
[3]: https://developer.mozilla.org/en-US/docs/Mozilla/JavaScript_code_modules/OSFile.jsm/OS.File_for_the_main_thread#write%28%29